code-studio/build-commands.js

```diff
@@ -49,10 +49,14 @@ exports.browserify = function (config) {
   var command = (shouldWatch ? 'watchify -v' : 'browserify') +
     (shouldMinify ? '' : ' --debug');
 
+  var reactifyStep = '-t reactify';
+
+  var globifyStep = '-t require-globify';
+
   var factorStep = '';
   if (shouldFactor) {
     factorStep = "-p [ factor-bundle -o '" + (shouldMinify ? 'uglifyjs ' : '') +
-      '>' + buildPath + "`basename $FILE .js`" + extension + "']";
+      '>' + buildPath + "`dirname $(realpath --relative-to " + srcPath + " $FILE)`/`basename -s .js -s .jsx $FILE`" + extension + "']";
```

Comment:
Agreed - I'll break this out somehow.  All I want to do is a fairly clever string transformation, so something like `sed` or even bash string replacement might be sufficient here.  In fact, what I've got right now, `basename -s .js -s .jsx $FILE` doesn't work because `basename` only takes one -s argument.

I'll fix this, at least extract it to its own line, and comment what it's doing really well.

I don't want us to be bound to this forever, but I'm really enjoying having every step expressed entirely as a shell command that we log, so you can easily re-run any part.  I feel like it takes a lot of the mystery out of the build.